### PR TITLE
Local development & deployment workflow

### LINKED ISSUE #15: runtime: I want to be able to run the server locally and force it to serve certain data
# Feature Request

Packages to improve:

- [X] runtime

### Description

As I develop a widget for consuming Docmaps, it would be helpful to be able to run the server locally and force it to serve certain Docmaps. This would be helpful so I can verify that my widget successfully fetches, parses, and displays data of various shapes and sizes from the server.

It's not that important to me how I provide this data to the server, as long as it's easy to create and update new fixtures.

### Use case

I am developing a package that interacts with the API server, and want to do local development.

### Proposed solution

N/A

### Additional information

N/A

### PR DESCRIPTION
**Pull request by ships**
WIP. will request review when ready.

## Description

Adds docker-compose based way to run the server and a useable backend on your local machine.

See the README.md in repository root for instructions. If you want, check out this branch and try it out yourself to see if it meets the needs of #15 .

### Related Issues

Resolves #15 .

### Checklist

- [x] I have tested these changes locally and they work as expected.
- [ ] I have added or updated tests to cover any new functionality or bug fixes.
- [x] I have updated the documentation to reflect any changes or additions to the project.
- [x] I have followed the [project's code of conduct](/CODE_OF_CONDUCT.md) and conventions for commit messages.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,4 +13,46 @@ See individual package READMEs for more info. In general, this monorepo depends
...
+of "similar seeming" blank nodes, and will result in duplication in the dataset. Named nodes will
+not be duplicated. For example, Steps and Actions are usually blank nodes, but Docmaps have ids.
+This means that if you upload a typical json-ld Docmap twice, all the steps will be doubled. If an
+action has an ID, then every copy of a step will point to it, whereas if not, there will be a copy
+of the action.
+
+To upload a JSONLD file:
+
+```bash
+<my_docmap.jsonld npx tsx scripts/upload_to_local_deployment.ts
```
Comment (andrewedstrom) [inline README.md:57]: I'm a little thrown off by this syntax—my bash is a bit rusty. By `<my_docmap.jsonld` here, are you saying I should literally do exactly this syntax, replacing `my_docmap.jsonld` with the exact filename of my docmap? 

e.g. `</tmp/docmaps/andrews_docmap.jsonld`? Or do I need to cat the contents of my docmap and then pipe it to `npx tsx scripts/upload_to_local_deployment.ts`?

Comment (andrewedstrom) [reply README.md:57]: Answering my own question, it looks like this command worked:

`</tmp/a-docmap.jsonld npx tsx scripts/upload_to_local_deployment.ts`

Comment (ships) [reply README.md:57]: @andrewedstrom , `</some/fd script` is the same stdin redirect you are accustomed to from the format `script < /some/fd` and is therefore effectively the same as `cat /some/fd | script` except without the extra process spawn :) i conventionally put it at the beginning so it reads left-to-right like a pipe but could change it.

**Review by andrewedstrom — APPROVED** [review]
Looks good!, I bet it was a lot of work to put this one together!

One suggestion for the new readme section:

It would be nice to have one example docmap committed to the repo, so the readme can give instructions for uploading one docmap from a known location, and then add a curl command showing how to retrieve the uploaded docmap.

File: README.md (hunk 1 above)
Comment (andrewedstrom) [reply README.md:57]: Fascinating! I slightly prefer writing it `script < /some/fd` or `cat ...` but that's really just a matter of what I happen to be familiar with. Doesn't necessarily need changing.

**Pull request merged**